### PR TITLE
[Feature]Support get paimon table createtime

### PR DESCRIPTION
**Pull request by leoyy0316**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [x] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Review by miomiocat — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 7 / 25 (28.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/PaimonScanNode.java|0|1|00.00%|[144]|
|:large_blue_circle:|com/starrocks/connector/paimon/PaimonMetadata.java|5|22|22.73%|[223, 256, 257, 258, 259, 260, 261, 262, 263, 264, 265, 266, 267, 268, 270, 271, 274]|
|:large_blue_circle:|com/starrocks/catalog/PaimonTable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 7 / 25 (28.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/PaimonScanNode.java|0|1|00.00%|[144]|
|:large_blue_circle:|com/starrocks/connector/paimon/PaimonMetadata.java|5|22|22.73%|[223, 256, 257, 258, 259, 260, 261, 262, 263, 264, 265, 266, 267, 268, 270, 271, 274]|
|:large_blue_circle:|com/starrocks/catalog/PaimonTable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/main/java/com/starrocks/connector/paimon/PaimonMetadata.java [hunk 1]
```diff
@@ -243,4 +254,35 @@ private List<Predicate> extractPredicates(PaimonTable paimonTable, ScalarOperato
...
+                    return updateTime.getMillisecond();
+                }
+            }
+        } catch (Exception e) {
+            LOG.error("Get paimon table {}.{} createtime failed, error: {}", dbName, tblName, e);
+        } finally {
+            if (iterator != null) {
+                iterator.close();
+            }
+            if (recordReader != null) {
```
Comment (miomiocat) [inline fe/fe-core/src/main/java/com/starrocks/connector/paimon/PaimonMetadata.java:282]: `iterator.close()` will call `recordReader.close()`, this step can be skipped

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 13 / 31 (41.94%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/PaimonScanNode.java|0|1|00.00%|[144]|
|:large_blue_circle:|com/starrocks/connector/paimon/PaimonMetadata.java|11|28|39.29%|[175, 176, 228, 263, 264, 265, 266, 267, 268, 269, 270, 271, 272, 273, 275, 280, 283]|
|:large_blue_circle:|com/starrocks/catalog/PaimonTable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 19 / 33 (57.58%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/planner/PaimonScanNode.java|0|1|00.00%|[145]|
|:large_blue_circle:|com/starrocks/connector/paimon/PaimonMetadata.java|17|30|56.67%|[176, 177, 229, 266, 273, 274, 275, 276, 277, 278, 279, 281, 286]|
|:large_blue_circle:|com/starrocks/catalog/PaimonTable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 19 / 31 (61.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/paimon/PaimonMetadata.java|17|29|58.62%|[176, 177, 271, 278, 279, 280, 281, 282, 283, 284, 286, 291]|
|:large_blue_circle:|com/starrocks/catalog/PaimonTable.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by miomiocat — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Comment by miomiocat** [comment]
@mergify backport-3.1

**Comment by miomiocat** [comment]
@mergify backport branch-3.1

**Comment by miomiocat** [comment]
@mergify backport branch-3.2

**Pull request merged**